File: App/Classes/Hub.m
#import "Hub.h"

@implementation Hub

@synthesize leftFlangeDiameter, leftFlangeToCenter;
@synthesize rightFlangeDiameter, rightFlangeToCenter, rear, holeCount;

+ (NSDictionary *) dataMap {
    return [NSDictionary dictionaryWithObjectsAndKeys:
            [NSArray arrayWithObjects:@"pk", @"integer", nil], @"id",
            [NSArray arrayWithObjects:@"createdAt", @"datetime", nil], @"created_at",
            [NSArray arrayWithObjects:@"updatedAt", @"datetime", nil], @"updated_at",
            [NSArray arrayWithObjects:@"brand", @"string", nil], @"brand",
            [NSArray arrayWithObjects:@"description", @"string", nil], @"description",
            [NSArray arrayWithObjects:@"leftFlangeDiameter", @"float", nil], @"left_flange_diameter",
            [NSArray arrayWithObjects:@"leftFlangeToCenter", @"float", nil], @"left_flange_to_center",
            [NSArray arrayWithObjects:@"rightFlangeDiameter", @"float", nil], @"right_flange_diameter",
            [NSArray arrayWithObjects:@"rightFlangeToCenter", @"float", nil], @"right_flange_to_center", 
            [NSArray arrayWithObjects:@"rear", @"boolean", nil], @"rear", 
            [NSArray arrayWithObjects:@"holeCount", @"integer", nil], @"hole_count", 
            nil];
}

- (void) dealloc {
    [super dealloc];
}

@end

File: rails/parts/app/models/hub.rb
class Hub < ActiveRecord::Base

  validates_presence_of :brand, :description, :left_flange_diameter, :left_flange_to_center, :right_flange_diameter, :right_flange_to_center, :hole_count
  validates_inclusion_of :rear, :in => [true, false]

  def self.from_csv(row)
    rear = row[3].to_f == 0.0
    offset = rear ? 5 : 0
    hub = Hub.new :part_number => row[0],
                  :brand => row[2].strip,
                  :description => row[1].strip,
                  :rear => rear,
                  :left_flange_diameter => row[3 + offset].to_f,
                  :right_flange_diameter => row[4 + offset].to_f,
                  :left_flange_to_center => row[5 + offset].to_f,
                  :right_flange_to_center => row[6 + offset].to_f,
                  :hole_count => row[7 + offset],
                  :verified => true

    hub.description.gsub! /^#{hub.brand}\W+/, ''
    hub
  end

end
